src/tests-2.1/pox-2.ts

```diff
@@ -0,0 +1,405 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import type { TestEnvContext } from './env-setup';
+import { ApiServer } from '../api/init';
+import * as supertest from 'supertest';
+import { Server } from 'net';
+import { DbBlock, DbEventTypeId, DbStxLockEvent, DbTx, DbTxStatus } from '../datastore/common';
+import { AnchorMode, bufferCV, makeContractCall, tupleCV, uintCV } from '@stacks/transactions';
+import { CoreRpcPoxInfo, StacksCoreRpcClient } from '../core-rpc/client';
+import { testnetKeys } from '../api/routes/debug';
+import * as poxHelpers from '../pox-helpers';
+import { PgWriteStore } from '../datastore/pg-write-store';
+import { StacksNetwork } from '@stacks/network';
+import * as btcLib from 'bitcoinjs-lib';
+import {
+  AddressStxBalanceResponse,
+  BurnchainRewardListResponse,
+  BurnchainRewardSlotHolderListResponse,
+  BurnchainRewardsTotal,
+} from '@stacks/stacks-blockchain-api-types';
+
+describe('PoX-2 tests', () => {
+  let db: PgWriteStore;
+  let api: ApiServer;
+  let client: StacksCoreRpcClient;
+  let stacksNetwork: StacksNetwork;
+
+  beforeAll(async () => {
+    const testEnv: TestEnvContext = (global as any).testEnv;
+    ({ db, api, client, stacksNetwork } = testEnv);
+    await Promise.resolve();
+  });
+
+  function standByForTx(expectedTxId: string): Promise<DbTx> {
+    return new Promise<DbTx>(resolve => {
+      const listener: (txId: string) => void = async txId => {
+        if (txId !== expectedTxId) {
+          return;
+        }
+        const dbTxQuery = await api.datastore.getTx({ txId: txId, includeUnanchored: false });
+        if (!dbTxQuery.found) {
+          return;
+        }
+        api.datastore.eventEmitter.removeListener('txUpdate', listener);
+        resolve(dbTxQuery.result);
+      };
+      api.datastore.eventEmitter.addListener('txUpdate', listener);
+    });
+  }
```

Comment:
I think we should refactor these onto the test utils (in another PR when we have some time in the future), they're very useful and used in a bunch of test suites